master/internal/project/postgres_project.go

```diff
@@ -210,6 +210,20 @@ RetryLoop:
 				} else {
 					p.Key = *requestedKey
 				}
+				var isUsed bool
+				err = tx.NewSelect().Table("local_id_redirect").
+					ColumnExpr("COUNT(*) > 0 as is_used").
+					Where("project_key = ?", p.Key).
+					Scan(ctx, &isUsed)
```

Comment:
Not that it's likely to make a huge difference, but I recommend the following query (or its Bun equivalent), instead:
```sql
SELECT EXISTS (
    SELECT 1 FROM local_id_redirect WHERE project_key = ?
) AS is_used
```
The query planner shows this being much faster:
```
determined=# explain SELECT EXISTS(SELECT 1 from experiments);
                              QUERY PLAN                              
----------------------------------------------------------------------
 Result  (cost=0.05..0.06 rows=1 width=1)
   InitPlan 1 (returns $0)
     ->  Seq Scan on experiments  (cost=0.00..12.80 rows=280 width=0)
(3 rows)

determined=# explain SELECT COUNT(*) > 0 FROM experiments;
                             QUERY PLAN                             
--------------------------------------------------------------------
 Aggregate  (cost=13.50..13.51 rows=1 width=1)
   ->  Seq Scan on experiments  (cost=0.00..12.80 rows=280 width=0)
(2 rows)
```
The high-level intuition I have about this is that `SELECT EXISTS(...` allows the database to give up looking for more once it's found at least one match whereas `COUNT(*) > 0` forces it to count every one and then do the comparison.

Reply:
Updated here a below where a similar query is made.